File: src/main/java/com/staticvoid/redtech/Redtech.java
package com.staticvoid.redtech;

import com.staticvoid.redtech.registry.RedtechBlocks;
import com.staticvoid.redtech.registry.RedtechItems;
import net.fabricmc.api.ModInitializer;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

@SuppressWarnings("unused")
public class Redtech implements ModInitializer {
    public static final String MOD_NAME = "Redtech";
    public static final String MOD_ID = "redtech";
    public static final String MOD_VERSION = "1.0.0-alpha.1";

    public static final Logger LOGGER = LoggerFactory.getLogger(MOD_NAME);

    @Override
    public void onInitialize() {
        RedtechBlocks.init();
        RedtechItems.init();

        LOGGER.info("{} v{} has been initialized.", MOD_NAME, MOD_VERSION);
    }
}

File: src/main/kotlin/com/staticvoid/redtech/Redtech.kt
package com.staticvoid.redtech

import com.staticvoid.redtech.registry.BlockRegistrar
import com.staticvoid.redtech.registry.ItemRegistrar
import org.slf4j.Logger
import org.slf4j.LoggerFactory

const val MOD_NAME = "Redtech"
const val MOD_ID = "redtech"
const val MOD_VERSION = "1.0.0-alpha.1"

val logger: Logger = LoggerFactory.getLogger(MOD_NAME)

fun init() {
    BlockRegistrar.init()
    ItemRegistrar.init()

    logger.info("$MOD_NAME version $MOD_VERSION has been initialized.")
}

fun initClient() {

}